spec/food_order_spec.rb

```diff
@@ -0,0 +1,56 @@
+require "food_order"
+require "menu"
+require "text_client"
+
+describe FoodOrder do
+
+  let(:dish_list) { { "test dish1" => 6.00, "test dish2" => 4.00, "test dish3" => 4.00 } } 
+  let(:menu_double) { instance_double(Menu, menu: dish_list) }
+
+  let(:formatted_time) { (Time.now + 3600).strftime("%k:%M") }
+  let(:formatted_confirmation) { "Thank you! Your order was placed and will be delivered before #{formatted_time}" }
```

Comment:
You can use the `timecop` gem to stub the time in your tests instead of relying on `Time.now`.
For example, you could freeze the time to Oct 1st 2pm, and expect the confirmation to be:
"Thank you! Your order was placed and will be delivered before 15:00". It's always better in your tests to control the environment as much as possible.